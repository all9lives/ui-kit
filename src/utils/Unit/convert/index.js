export default unit => `${unit / 16}rem`
