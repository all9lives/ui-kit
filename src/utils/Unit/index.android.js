const convert = unit => unit

export default {
  convert
}
