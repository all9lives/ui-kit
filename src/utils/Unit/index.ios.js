import Unit from './index.android';

export default Unit;
