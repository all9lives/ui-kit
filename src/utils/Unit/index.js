import _ from 'lodash'
import convert from './convert'

const parse = (input, parser = _.identity) => {
  if (typeof input === 'number') {
    return convert(input, parser)
  }
  const units = input.split(' ')
  return units.map(unit => convert(unit, parser)).join(' ')
}

export default {
  convert,
  parse
}
