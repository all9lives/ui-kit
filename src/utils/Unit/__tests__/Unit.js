/* eslint-env jest */
import Unit from '../index.js'

jest.mock('../convert', () => (n, parser) => `proper-${parser(n)}`)

it('parses an number input into a proper output', () => {
  expect(Unit.parse(16)).toMatchSnapshot()
})

it('parses a string input into a proper output', () => {
  expect(Unit.parse('16 16')).toMatchSnapshot()
})

it('parses a string input with a parser into a proper output', () => {
  expect(Unit.parse('16 16', n => `parsed-${n}`)).toMatchSnapshot()
})
