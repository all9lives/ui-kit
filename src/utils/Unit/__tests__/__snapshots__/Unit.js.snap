// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parses a string input into a proper output 1`] = `"proper-16 proper-16"`;

exports[`parses a string input with a parser into a proper output 1`] = `"proper-parsed-16 proper-parsed-16"`;

exports[`parses an number input into a proper output 1`] = `"proper-16"`;
